entrypoint.sh

```diff
@@ -1,187 +1,21 @@
-uid=$(id -u)
-gid=$(id -g)
-# turn off -e for getent because it will return error code in anonymous uid case
-set +e
-uid_entry=$(getent passwd ${uid})
-set -e
+declare SPARK_HISTORY_OPTS=""
 
-# If there is no passwd entry for the container UID, attempt to create one
-if [[ -z "${uid_entry}" ]] ; then
-    if [[ -w /etc/passwd ]] ; then
-        echo "$uid:x:$uid:$gid:anonymous uid:${SPARK_HOME}:/bin/false" >> /etc/passwd
-    else
-        echo "Container entrypoint.sh failed to add passwd entry for anonymous UID"
-    fi
-fi
+readonly eventsDir="${1}"
+readonly serverSideEncryption="${2}"
 
-if [[ "$enablePVC" == "true" ]]; then
-    export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-    -Dspark.history.fs.logDirectory=file:/mnt/$eventsDir";
-elif [[ "$enableGCS" == "true" ]]; then
-    if [[ -n ${gcloudKey} ]]; then
-      export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-      -Dspark.hadoop.google.cloud.auth.service.account.json.keyfile=/etc/secrets/$gcloudKey \
-      -Dspark.history.fs.logDirectory=$eventsDir";
-    else
-      echo "Please pass your Google Cloud Key!"
-      exit 1
-    fi
-elif [[ "$enableS3" == "true" ]]; then
-    export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-      -Dspark.history.fs.logDirectory=$eventsDir
-      -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem";
-    if [[ "$enableIAM" == "false" ]]; then
-      export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-      -Dspark.hadoop.fs.s3a.access.key=$(cat /etc/secrets/${accessKeyName}) \
-      -Dspark.hadoop.fs.s3a.secret.key=$(cat /etc/secrets/${secretKeyName})";
-    fi;
-elif [ "$enableWASBS" == "true" ]; then
-  export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-    -Dspark.history.fs.logDirectory=$eventsDir \
-    -Dspark.hadoop.fs.defaultFS=wasbs://$containerName@$storageAccountName.blob.core.windows.net \
-    -Dspark.hadoop.fs.wasbs.impl=org.apache.hadoop.fs.azure.NativeAzureFileSystem \
-    -Dspark.hadoop.fs.AbstractFileSystem.wasbs.impl=org.apache.hadoop.fs.azure.Wasbs";
-  if [ "$sasKeyMode" == "true" ]; then
-    export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-      -Dspark.hadoop.fs.azure.local.sas.key.mode=true \
-      -Dspark.hadoop.fs.azure.sas.$containerName.$storageAccountName.blob.core.windows.net=$(cat /etc/secrets/${sasKeyName})";
-  else
-    export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-      -Dspark.hadoop.fs.azure.account.key.$storageAccountName.blob.core.windows.net=$(cat /etc/secrets/${storageAccountKeyName})";
-  fi;
-else
-    export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
-    -Dspark.history.fs.logDirectory=$eventsDir";
-fi;
+export SPARK_HISTORY_OPTS="$SPARK_HISTORY_OPTS \
+  -Dspark.history.fs.logDirectory=${eventsDir} \
+  -Dspark.hadoop.fs.s3a.impl=org.apache.hadoop.fs.s3a.S3AFileSystem \
+  -Dspark.hadoop.fs.s3a.server-side-encryption-algorithm=${serverSideEncryption} \
+  -Dspark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.EnvironmentVariableCredentialsProvider";
```

Comment:
Why not use the default credential provider chain?
```suggestion
  -Dspark.hadoop.fs.s3a.aws.credentials.provider=com.amazonaws.auth.DefaultAWSCredentialsProviderChain";
```
BTW isn't it the default?

Reply:
Nope, I don't want to use the default. The default is simple with access and secret keys. But we can go with a secret object and environment in this way.

Reply:
The default provider chain chains all the others and `EnvironmentVariableCredentialsProvider` is the first in the chain:
https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/credentials.html#credentials-default